Src/FluentAssertions/Common/IClock.cs

```diff
@@ -7,8 +7,10 @@ namespace FluentAssertions.Common
     /// <summary>
     /// Represents an abstract timer that is used to make some of this library's timing dependent functionality better testable.
     /// </summary>
-    public interface ITimer
+    public interface IClock
```

Comment:
In a similar "challenge" it was named `ITimerFactory`.
I think, "clock" is a very generic name.
Maybe, it could also be named `IAssertionTiming` and `IAssertionTimer`.

Reply:
I named it like that because it does more than just producing timers now.

Reply:
All is related to **timing**, this is why I recommend IAssertion**Timing**.

Reply:
I kept `IClock`